[1/n][torch/elastic] Move torchelastic docs *.rst

kiukchung opened the pull request:
Summary: Moves docs sphinx `*.rst` files from the torchelastic repository to torch. Note: only moves the rst files the next step is to link it to the main pytorch `index.rst` and write new `examples.rst`

Differential Revision: D27974751

H-Huang reviewed (approved):
Lint error updated on internal diff, stamping to unblock

Commit:
[1/n][torch/elastic] Move torchelastic docs *.rst (#148)

Summary:
Pull Request resolved: https://github.com/pytorch/elastic/pull/148

Pull Request resolved: https://github.com/pytorch/pytorch/pull/56811

Moves docs sphinx `*.rst` files from the torchelastic repository to torch. Note: only moves the rst files the next step is to link it to the main pytorch `index.rst` and write new `examples.rst`

Reviewed By: H-Huang

Differential Revision: D27974751

fbshipit-source-id: aabf8f716ea723a50373a8dd02a3909e4af571fc
